### PR TITLE
samples: sof: Moved software configuration to SOF

### PR DESCRIPTION
**Pull request by aborisovich**
Existing SOF  platforms software configuration had been moved to SOF repository in form of the overlays in [PR5531: Moved platforms software configuration from Zephyr to SOF](https://github.com/thesofproject/sof/pull/5531).
Deleting duplicates.

Signed-of-by: Andrey Borisovich <andrey.borisovich@intel.com>

### CODE REVIEW

**Comment by marc-hb** [comment]
@aborisovich this should help:
- https://github.com/zephyrproject-rtos/zephyr/issues/38725

In the mean time just cc: people in a comment like you just did.

Can you please submit this to SOF CI using the one-line hack in sof/scripts/xtensa-build-zephyr.py?

**Review by nashif — CHANGES_REQUESTED** [review]
not sure how all of this is going to work, now we are left with a sample without any configurations in the zephyr tree and no way to build it.  
What you need to do is move the complete application/sample into SOF and have all the configurations maintained in SOF, not as overlays and point to that sample in the module.yml file. The least we want is some sanitycheck that SOF build with Zephyr without too much magic, similar to what we have right now.

**Comment by aborisovich** [comment]
> not sure how all of this is going to work, now we are left with a sample without any configurations in the zephyr tree and no way to build it. What you need to do is move the complete application/sample into SOF and have all the configurations maintained in SOF, not as overlays and point to that sample in the module.yml file. The least we want is some sanitycheck that SOF build with Zephyr without too much magic, similar to what we have right now.

I am in the crossroads here.
So the initial idea was to move the SOF application as whole to the SOF repo but we wanted to leave a board definitions in the Zephyr. With this idea overlays must stay as we use them to append "software configuration" to the board definitions.
I did not moved a whole zephyr application in the single PR to separate the work to atomic steps.

What blocks me is twister CI.
Is there any expert on this here that would help me?
What the twister is doing now - it creates a .csv test plan using --all flag (all platforms).
This flag can not be used with my test case as we need to add twister --extra-args=OVERLAY_CONFIG=./../../../../../modules/audio/sof/overlays/$PLATFORM/overlay.conf
How can I create a complete .csv test plan for all platforms by invoking one by one (per platform) commands to twister?

Long story short I need something like:
<for each platform>
./zephyr/scripts/twister -c -T ./zephyr/samples/subsys/audio/sof --platform <board_name> --extra-args=OVERLAY_CONFIG=./modules/audio/sof/overlays/<sof_platform_name>/overlay.conf  --save-tests _test_plan_partial.csv

@andyross @nashif who can I ask for help as experienced CI person with Zephyr?
This change is not a temporary shot as we decided to split hardware and software configuration to board and overlays as far as I remember our technical discussions.

**Comment by marc-hb** [comment]
Not sure how "temporary" this would be but this should work:

```
samples/subsys/audio/sof
├── intel_adsp_cavs15
│   ├── CMakeLists.txt
│   └── sample.yml
├── intel_adsp_cavs18
│   ├── CMakeLists.txt
│   └── sample.yml
└── nxp_adsp_imx8
    ├── CMakeLists.txt
    └── sample.yml
```

Then you can "hardcode" each `sample.yml` to its corresponding platform with `extra_args:` https://docs.zephyrproject.org/latest/develop/test/twister.html

Once the `extra_args` is not required anymore then maybe it can be collapsed back into a single example (leave plenty comments about that)

My 2 cents.

**Comment by nashif** [comment]
> I did not moved a whole zephyr application in the single PR to separate the work to atomic steps.

I think this is the wrong approach. You should move it all or leave it as is. I do not see any reason why this should be done in steps and having to go through workarounds to make this happen.

**Comment by nashif** [comment]
> What blocks me is twister CI.

this is just part of it, you are leaving an application in zephyr that cannot be built and without any configuration. Before getting into twister and CI the question is, how can I build this sample in zephyr?

**Comment by aborisovich** [comment]
> > I did not moved a whole zephyr application in the single PR to separate the work to atomic steps.
> 
> I think this is the wrong approach. You should move it all or leave it as is. I do not see any reason why this should be done in steps and having to go through workarounds to make this happen.

Ok, I will then drop this PR and start moving sof application to sof repo.
Then we will submit single "clean up" PR deleting duplicates.

> this is just part of it, you are leaving an application in zephyr that cannot be built and without any configuration. Before getting into twister and CI the question is, how can I build this sample in zephyr using west or cmake directly.?

@nashif its not like that. You can build application using west just like previously. Nothing changed except extra flag.
You can not build sample sof application without sof module anyway as you need its source code and CMake file.. The fact that I move configuration there changes nothing but the extra flag to west build command. But ok I get your point.

**Comment by aborisovich** [comment]
> Not sure how "temporary" this would be but this should work:
> 
> ```
> samples/subsys/audio/sof
> ├── intel_adsp_cavs15
> │   ├── CMakeLists.txt
> │   └── sample.yml
> ├── intel_adsp_cavs18
> │   ├── CMakeLists.txt
> │   └── sample.yml
> └── nxp_adsp_imx8
>     ├── CMakeLists.txt
>     └── sample.yml
> ```
> 
> Then you can "hardcode" each `sample.yml` to its corresponding platform with `extra_args:` https://docs.zephyrproject.org/latest/develop/test/twister.html
> 
> Once the `extra_args` is not required anymore then maybe it can be collapsed back into a single example (leave plenty comments about that)
> 
> My 2 cents.

Thanks for the help.
This will be useful now in other PR when we move sof application to sof repo as we need to use overlays anyway,
